tests/Gedmo/Uploadable/UploadableEntityTest.php

```diff
@@ -377,19 +377,24 @@ public function testFileWithFilenameAlphanumericGenerator()
 
     public function testFileWithCustomFilenameGenerator()
     {
-        $file = new FileWithCustomFilenameGenerator();
-        $fileInfo = $this->generateUploadedFile();
+        $uploadFile = function($expected) {
+            $file = new FileWithCustomFilenameGenerator();
+            $fileInfo = $this->generateUploadedFile();
 
-        $this->listener->addEntityFileInfo($file, $fileInfo);
+            $this->listener->addEntityFileInfo($file, $fileInfo);
 
-        $this->em->persist($file);
-        $this->em->flush();
+            $this->em->persist($file);
+            $this->em->flush();
 
-        $this->em->refresh($file);
+            $this->em->refresh($file);
 
-        $filename = substr($file->getFilePath(), strrpos($file->getFilePath(), '/') + 1);
+            $filename = substr($file->getFilePath(), strrpos($file->getFilePath(), '/') + 1);
+
+            $this->assertEquals($expected, $filename);
+        };
 
-        $this->assertEquals('123.txt', $filename);
+        $uploadFile('123.text');
```

Comment:
If we need to perform the same test through multiple cases, I think we should use a data provider instead.